examples/playwright/test/basic.test.ts

```diff
@@ -0,0 +1,40 @@
+import { afterAll, beforeAll, describe, expect, test } from 'vitest'
+import { preview } from 'vite'
+import type { PreviewServer } from 'vite'
+import * as playwright from 'playwright'
+import type { Browser, BrowserContext, Page } from 'playwright'
+import type { AddressInfo } from 'net'
+
+describe('basic', async() => {
+  let server: PreviewServer
+  let browser: Browser
+  let context: BrowserContext
+  let page: Page
+
+  beforeAll(async() => {
+    server = await preview({ preview: { port: 3000 } })
+    // Might be a different one if port 3000 is already in use.
+    const port = (server.httpServer.address() as AddressInfo).port;
+    browser = await playwright.chromium.launch()
```

Comment:
I was looking here for an option inside the test-runner to run all the tests parameterized. In this case for:

- webkit
- firefox
- chromium

So it would start in the end 3 workers for it. Is something like this possible? I only saw `describe.each`.

Reply:
Playwright's own test-runner allows something like that and then calculates based on that the amount of workers for concurrency accordingly:
```js
const config = {
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'], someKey: 'foobar1' },
    },
    {
      name: 'firefox',
      use: { ...devices['Desktop Firefox'], someKey: 'foobar2' },
    },
    {
      name: 'webkit',
      use: { ...devices['Desktop Safari'], someKey: 'foobar3' },
    },
  ],
};
```

So `use` is the configuration then for the given project.

Yeah I'll update the PR for now so its based on the env.

Reply:
I think @dominikg is working on a plugin for playwrite testing, maybe you can collaborate? It seems the feature is rather usefull